src/app/children/child-details/child-details.component.ts

```diff
@@ -170,4 +195,36 @@ export class ChildDetailsComponent implements OnInit {
   navigateBack() {
     this.location.back();
   }
+
+  addHealthCheck(date: Date, height: number, weight: number){
+    var newHealthCheck = new HealthCheck(uniqid());
+    newHealthCheck.date=date;
+    newHealthCheck.height=height;
+    newHealthCheck.weight=weight;
+    newHealthCheck.child=this.child.getId();
+    console.log(newHealthCheck);
+    this.entityMapperService.save<HealthCheck>(newHealthCheck);
+    this.loadHealthChecks();
+  }
+
+  loadHealthChecks(){
+
+    let tempArray = []; //we need this because somehow you cant push directly into the HealthCheckRecords Array
+    //this is a workaround until indizes in our database are centrelized 
+    this.entityMapperService.loadType<HealthCheck>(HealthCheck).then(
+       result => result.forEach(doc => {
+           if(doc.child===this.child.getId()){
+                tempArray.push(doc);
+            }
+            else{
+              console.log(doc);
+            }
+       })
+       ).catch(error => console.log("ERROR FROM LOADHEALTHCHECKS"));
+       console.log(tempArray);
+       this.HealthCheckRecords=tempArray;
```

Comment:
Try putting the `this.HealthCheckRecords=tempArray;` inside the `then(...)` block above. (Can't test it because I am getting other errors)

Async can be a mess: whatever is inside the `then(...)` is only called later, so when your code `this.HealthCheckRecords=tempArray;` that is outside the `then(...)` is called immediately, the tempArray is still empty. And when the tempArray is finally filled, you are not assigning it to this.HealthCheckRecords again.